src/test/java/io/quarkus/gizmo2/FieldAccessTest.java

```diff
@@ -110,4 +112,74 @@ public void testConstantField() {
         });
         assertEquals(7, tcm.instanceMethod("test", ToIntFunction.class).applyAsInt(tcm.constructor(Supplier.class).get()));
     }
+
+    @Test
+    public void testInstanceConstantField() {
+        TestClassMaker tcm = new TestClassMaker();
+        Gizmo g = Gizmo.create(tcm);
+        String className = "io.quarkus.gizmo2.InstanceConstantFieldTest";
+        g.class_(className, zc -> {
+            // create an instance field and initialize it to a constant value
+            FieldDesc field1 = zc.field("field1", ifc -> {
+                ifc.withType(String.class);
+                ifc.final_();
+                ifc.withInitial("Hello world");
+            });
+            // create a constructor that does not explicitly initialize the field
+            zc.constructor(cc -> {
+                Var this_ = cc.this_();
+                cc.public_();
+                cc.body(b0 -> {
+                    b0.invokeSpecial(ConstructorDesc.of(Object.class), this_);
+                    b0.return_();
+                });
+            });
+            // a test method to verify the result
+            zc.staticMethod("test0", smc -> {
+                smc.returning(boolean.class);
+                smc.public_();
+                smc.body(b0 -> {
+                    Expr instance = b0.new_(ClassDesc.of(className));
+                    b0.return_(b0.exprEquals(instance.field(field1), Constant.of("Hello world")));
+                });
+            });
+        });
+        assertTrue(tcm.staticMethod("test0", BooleanSupplier.class).getAsBoolean());
+    }
+
+    @Test
+    public void testInstanceInitializedField() {
+        TestClassMaker tcm = new TestClassMaker();
+        Gizmo g = Gizmo.create(tcm);
+        String className = "io.quarkus.gizmo2.InstanceInitializedFieldTest";
+        g.class_(className, zc -> {
+            // create an instance field and initialize it to a non-constant value
+            FieldDesc field1 = zc.field("field1", ifc -> {
+                ifc.withType(String.class);
+                ifc.final_();
+                ifc.withInitializer(b0 -> {
+                    b0.yield(b0.withString(Constant.of("Hello")).concat(Constant.of(" world")));
```

Comment:
The `yield()` call here is not entirely intuitive, but I'm not sure what we can do about that. Maybe make `withInitializer()` take `Function<BlockCreator, Expr>` instead of `Consumer<BlockCreator>`?

Reply:
This is not specific to this case... any block which has an output type (including switch expressions, `?:`, etc.) works this way. I agree it's not super intuitive and it can be easy to mess it up, but I don't see a good way of changing it in a uniform way without breaking a lot of things (or else changing *all* `Consumer` to `Function` and requiring users to return a `void` constant or something).

Reply:
We can have a separate discussion on that, at any rate.